### PR TITLE
feat!: unbounded blocksize test

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

This PR introduces the unbounded block size test that I've been using for the past few months to collect data on testground. I'll be going through this later today and removing any less permanent things that were here.

closes https://github.com/celestiaorg/celestia-core/issues/1140 and https://github.com/celestiaorg/celestia-core/issues/945

### CODE REVIEW

File: test/testground/manifest.toml [hunk 1]
```diff
@@ -38,9 +39,10 @@ blobs_per_sequence = { type = "int", default = 1 }
 inbound_peer_count = { type = "int", default = 40 }
 outbound_peer_count = { type = "int", default = 10 }
 gov_max_square_size = { type = "int", default = 256 }
-max_block_bytes = { type = "int", default = 100000000 }
+max_block_bytes = { type = "int", deafult = 100000000 }
```
Comment (rootulp) [inline test/testground/manifest.toml:42]: [typo]

```suggestion
max_block_bytes = { type = "int", default = 100000000 }
```

File: test/testground/network/configurators.go [hunk 2]
```diff
@@ -114,48 +113,30 @@ func ConnectRandom(numPeers int) Configurator {
...
-		if node.GroupID != SeedGroupID {
-			nodes[i].CmtConfig.P2P.Seeds = strings.Join([]string{seedNode.PeerID}, ",")
-		}
-	}
-	return nodes, nil
-}
-
+// TracingConfigurator is a Configurator that configures tracing for the
+// network. It will set the nodes to collect only the round state data, and will
+// set the nodes specified in the TracingParams to collect all trae data.
```
Comment (rootulp) [inline test/testground/network/configurators.go:118]: [typo]

```suggestion
// set the nodes specified in the TracingParams to collect all trace data.
```

File: test/testground/network/consensus_node.go [hunk 3]
```diff
@@ -35,7 +34,7 @@ import (
 
 // ConsensusNode is the node type used by testground instances to run a
 // celestia-app full node. It can optionally be configured to be a validator,
-// and has methods to bootstrap a network, initialize itself, start, and stop.
+// and has methods to boostrap a network, initialize itself, start, and stop.
```
Comment (rootulp) [inline test/testground/network/consensus_node.go:37]: [typo]

```suggestion
// and has methods to bootstrap a network, initialize itself, start, and stop.
```

File: test/testground/network/experiments.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	ConsistentFill     = "consistent-fill"
+)
+
+func init() {
+	// coretypes.SetBlockPartSizeBytes(3220000)
+	// consensus.SetMaxMsgSize(5000000)
+	// consensus.SetBlockPartPriority(50)
+	// consensus.SetRecvBufferCapacity(100 * 50 * 4096)
+	// consensus.SetSendQueueCapacity(10000)
+}
```
Comment (rootulp) [inline test/testground/network/experiments.go:26]: if `init` isn't necessary, proposal to remove

```suggestion
```

File: test/testground/network/sync.go [hunk 5]
```diff
@@ -132,15 +132,13 @@ func DownloadNodeConfigs(ctx context.Context, runenv *runtime.RunEnv, initCtx *r
 // GenesisWrapper is a simple struct wrapper that makes it easier testground to
```
Comment (rootulp) [inline test/testground/network/sync.go:132]: I think the comment removals in this file are accidental and should prob be reverted.

**Review by rootulp — COMMENTED** [review]
[question] I don't have a great solution but should we be concerned about the repetitive .toml configs? Is it possible to extract a common `.toml` config and then have the individual files only override the params that are unique for that test case? Perhaps that's over complicating the tests so feel free to ignore.

None of my feedback is strictly blocking b/c this is mostly modifying testground test code which is kinda :shipit: whenever you want

File: go.mod [hunk 6]
```diff
@@ -241,5 +241,5 @@ replace (
 	github.com/cosmos/ledger-cosmos-go => github.com/cosmos/ledger-cosmos-go v0.12.4
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 	github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
-	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.32.0-tm-v0.34.29
+	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.31.0-tm-v0.34.29
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply go.mod:244]: yeah per the note earlier I'm not sure what is best here. Besides fixing the toml bug in viper I don't see a solution that's good. We either have two different branches of core or do a hack for backwards compatible fields.

File: pkg/appconsts/versioned_consts.go [hunk 7]
```diff
@@ -3,11 +3,10 @@ package appconsts
 import (
 	"github.com/celestiaorg/celestia-app/pkg/appconsts/testground"
 	v1 "github.com/celestiaorg/celestia-app/pkg/appconsts/v1"
-	v2 "github.com/celestiaorg/celestia-app/pkg/appconsts/v2"
 )
 
 const (
-	LatestVersion = v2.Version
+	LatestVersion = v1.Version
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply pkg/appconsts/versioned_consts.go:9]: good catch. I think I was just being paranoid and wanted to make sure we were using as close to v1 as possible. 

reverted! 3ca930e37fe1d0fee46ac04bb79b075bf6c2cd9e

File: test/testground/compositions/standard/plan.toml [hunk 8]
```diff
@@ -1,39 +1,40 @@
...
 halt_height = "50"
 latency = "0"
+random_latency = "false"
 bandwidth = "1Gib"
 validators = "40"
 topology = "seed"
 pex = "true"
-timeout_propose = "10s"
-timeout_commit = "11s"
+timeout_propose = "120s"
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/testground/compositions/standard/plan.toml:22]: with these experiments, we're trying to see if specific changes have any effect on large blocks. Without raising this, the control and other experiments would all fail at or after 16MB, which wouldn't provide us with data on any effect on throughput for large blonks

File: test/testground/compositions/standard/plan.toml [hunk 9]
```diff
@@ -42,13 +43,14 @@ builder = "docker:generic"
 memory = "8Gi"
 cpu = "6"
 [groups.instances]
-count = 40
+count = 100
 percentage = 0.0
 [groups.build_config]
-build_base_image = "golang:1.21.6"
+build_base_image = "golang:1.21.3"
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/testground/compositions/standard/plan.toml:49]: idk, it might have been a weird merge thing since I've been using this branch a while

good catch!

File: test/testground/compositions/utils.go [hunk 10]
```diff
@@ -39,11 +41,19 @@ func CreateNetworkConfig(runenv *runtime.RunEnv, initCtx *run.InitContext) (netw
 	if err != nil {
 		return network.Config{}, err
 	}
+	l := runenv.IntParam("latency")
+	// rand.Intn will panic if l == 0
+	if l == 0 {
+		l = 1
+	}
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/testground/compositions/utils.go:48]: leaving this for now just cause an if statement is more robust, and debugging testground failures is #maxpain but I'm also not opposed to changing it

File: test/testground/compositions/standard/plan.toml (hunk 8 above)
[the remark this answers is not in the document]
Comment (cmwaters) [reply test/testground/compositions/standard/plan.toml:22]: But this file is under the standard directory not the unbounded one. I would have through standard would imply as close to mainnet as possible

File: test/testground/compositions/standard/plan.toml [hunk 11]
```diff
@@ -1,39 +1,40 @@
 [metadata]
-name = "standard"
```
Comment (cmwaters) [inline test/testground/compositions/standard/plan.toml:2]: Why did you remove this?

File: test/util/sdkutil/proposals.go [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+	blobtypes "github.com/celestiaorg/celestia-app/x/blob/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// MaxBlockBytesParamChange returns a param change for the max block bytes
+// consensus params.
+func MaxBlockBytesParamChange(codec codec.Codec, maxBytes int) proposal.ParamChange {
```
Comment (cmwaters) [inline test/util/sdkutil/proposals.go:15]: Could this be used in other places?

File: test/util/testnode/config.go [hunk 13]
```diff
@@ -124,8 +124,7 @@ func DefaultConfig() *Config {
 		WithAppConfig(DefaultAppConfig()).
 		WithAppOptions(DefaultAppOptions()).
 		WithAppCreator(cmd.NewAppServer).
-		WithSuppressLogs(true).
-		WithConsensusParams(DefaultConsensusParams())
```
Comment (cmwaters) [inline test/util/testnode/config.go:128]: Why remove this?

**Review by cmwaters — COMMENTED** [review]
Few questions

File: test/testground/manifest.toml (hunk 1 above)
Comment (evan-forbes) [reply test/testground/manifest.toml]: thanks good catch

File: test/testground/compositions/standard/plan.toml (hunk 8 above)
[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/testground/compositions/standard/plan.toml:22]: fair point, I will revert the altered timeouts in the standard compositions

File: test/util/testnode/config.go (hunk 13 above)
Comment (evan-forbes) [reply test/util/testnode/config.go:128]: due to working on this branch for too long, I don't remember, but added them back

File: test/util/sdkutil/proposals.go (hunk 12 above)
Comment (evan-forbes) [reply test/util/sdkutil/proposals.go:15]: yeah, we're using it in the sdk test and the square size test

File: test/testground/compositions/standard/plan.toml (hunk 11 above)
Comment (evan-forbes) [reply test/testground/compositions/standard/plan.toml:2]: it wasn't being used anywhere afaict. testground always refers to tests using a unique identifier

File: test/testground/network/experiments.go [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+		BlobCount:     count,
+	})
+
+	runenv.RecordMessage("leader: sending txsim command")
+
+	_, err := initCtx.SyncClient.Publish(ctx, CommandTopic, cmd)
+	return err
+}
+
+func (l *Leader) unboundedBlockSize(
```
Comment (staheri14) [inline test/testground/network/experiments.go:38]: Would be great to associate this function with some descriptions.

File: test/testground/network/experiments.go [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+	query := "tm.event = 'NewBlockHeader'"
+	events, err := l.cctx.Client.Subscribe(ctx, "leader", query, 10)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		blockSize := 2000000
+		blockIncrement := 5000000
```
Comment (staheri14) [inline test/testground/network/experiments.go:58]: A comment on the size unit (byte or bit) would be nice.

File: test/testground/network/sync.go [hunk 16]
```diff
@@ -195,10 +193,7 @@ func DownloadGenesis(ctx context.Context, initCtx *run.InitContext) (json.RawMes
 	return genesis, nil
 }
 
-// DownloadSync downloads the given topic from the sync service. It will
-// download the given number of messages from the topic. If the topic is closed
-// before the expected number of messages are received, an error is returned.
-func DownloadSync[T any](ctx context.Context, initCtx *run.InitContext, topic *sync.Topic, t T, count int) ([]T, error) {
+func DownloadSync[T any](ctx context.Context, initCtx *run.InitContext, topic *sync.Topic, _ T, count int) ([]T, error) {
```
Comment (staheri14) [inline test/testground/network/sync.go:196]: Was the removal of the go documentation intentional?

File: test/testground/network/follower.go [hunk 17]
```diff
@@ -87,7 +87,7 @@ func (f *Follower) Plan(ctx context.Context, runenv *runtime.RunEnv, initCtx *ru
 
 	runenv.RecordMessage("follower waiting for start height")
 
-	_, err = f.cctx.WaitForHeightWithTimeout(int64(10), time.Minute*7)
+	_, err = f.cctx.WaitForHeightWithTimeout(int64(5), time.Minute*7)
```
Comment (staheri14) [inline test/testground/network/follower.go:90]: [Question] Is the starting height value arbitrary, or is it determined by a specific logic within a part of the code? I'm curious about the rationale behind its change from `10` to `5.`

File: test/testground/compositions/standard/plan.toml [hunk 18]
```diff
@@ -1,39 +1,40 @@
...
 author = "core-app"
 
 [global]
-plan = "celestia"
+plan = "core-app"
```
Comment (staheri14) [inline test/testground/compositions/standard/plan.toml:5]: [Question] Does this change require updating the test instructions [here](https://github.com/celestiaorg/celestia-app/tree/main/test/testground#running-the-experiment) where the plan name is `celestia`?

File: test/testground/network/follower.go (hunk 17 above)
Comment (evan-forbes) [reply test/testground/network/follower.go:90]: iirc this was just an optimization as there's no benefit to waiting for block height 10. 5 is still a bit aribitrary

File: test/testground/network/experiments.go (hunk 15 above)
Comment (evan-forbes) [reply test/testground/network/experiments.go]: good idea, added here 05e325798ef09d1674701e9508ac8398e172d023

File: test/testground/network/experiments.go (hunk 14 above)
Comment (evan-forbes) [reply test/testground/network/experiments.go:39]: good idea, added here 0140a0a2ec0ee08994a50a2e96142f50c86d2c6c

File: test/testground/compositions/standard/plan.toml (hunk 18 above)
Comment (evan-forbes) [reply test/testground/compositions/standard/plan.toml:5]: idk, as I think I've done both and it worked, but good point and changed here 9ba189188e4711af942257bccf60d59cd301001c

File: test/testground/network/sync.go (hunk 16 above)
Comment (evan-forbes) [reply test/testground/network/sync.go:199]: no it wasn't good catch, I think was some issue with merging. fixed in 4bcb9d03d88e4bde3227feaff84a1fdae0b0c68c

File: test/testground/network/follower.go (hunk 17 above)
Comment (staheri14) [reply test/testground/network/follower.go:90]: I see, then why not 1?

Comment (evan-forbes) [reply test/testground/network/follower.go:90]: I don't have data to know how long this takes, but it does take some amount of time to establish connections between all the peers, so I guestimated 5

**Review by rootulp — DISMISSED** [review]
Approving to unblock. I haven't tried to run any of these tests

File: go.mod (hunk 6 above)
[the remark this answers is not in the document]
Comment (evan-forbes) [reply go.mod:244]: stop

also https://github.com/celestiaorg/celestia-core/issues/1197 resolves this comment

**Review by rootulp — APPROVED** [review]

File: test/testground/compositions/utils.go [hunk 19]
```diff
@@ -85,7 +95,7 @@ func parseBandwidth(s string) (uint64, error) {
 		return 0, fmt.Errorf("unknown unit in string: %s", s)
 	}
 
-	numberStr := strings.TrimRight(s, "KibMibGibTibKBMGBT")
+	numberStr := strings.TrimRight(s, "KMGTib")
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/testground/compositions/utils.go:98]: the only reason this was shortened was because of the linter. I can see how this isn't the most clear, but for now since this is just a testground helper, provided its working as intented I'm just going to leave it

[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/testground/compositions/utils.go:98]: if more than one reviewer thinks this is a good idea I'm happy to change

**Review by cmwaters — APPROVED** [review]
utAck

File: test/testground/compositions/utils.go (hunk 19 above)
[the remark this answers is not in the document]
Comment (rootulp) [reply test/testground/compositions/utils.go:98]: Not a big deal, can ship as is 🚢

**Review by staheri14 — APPROVED** [review]
LGTM (I haven't tested the code though)

File: test/testground/compositions/standard/plan.toml [hunk 20]
```diff
@@ -23,17 +23,18 @@ timeout_propose = "10s"
...
-gov_max_square_size = "128"
-max_block_bytes = "2000000"
+gov_max_square_size = "512"
+max_block_bytes = "99000000"
 mempool = "v1"
 broadcast_txs = "true"
 tracing_nodes = "0"
 tracing_token = ""
 tracing_url = ""
+experiment = "consistent-fill"
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/testground/compositions/standard/plan.toml:37]: @coderabbitai stop

**Review by staheri14 — APPROVED** [review]

**Comment by rootulp** [comment]
It's been so long that I totally forgot about this PR. It would help if the PR description stated the breaking changes included in this PR. Also does it a "breaking" label? 

https://github.com/celestiaorg/celestia-core/issues/1140 seems no longer relevant because it's already closed.

File: test/testground/go.mod [hunk 21]
```diff
@@ -34,6 +34,8 @@ require (
 	github.com/bits-and-blooms/bitset v1.10.0 // indirect
 	github.com/btcsuite/btcd/btcec/v2 v2.3.2 // indirect
 	github.com/celestiaorg/blobstream-contracts/v3 v3.1.0 // indirect
+	github.com/celestiaorg/go-square v1.0.0-rc0 // indirect
```
Comment (rootulp) [inline test/testground/go.mod:37]: [no change needed] 🤔 we should bump to v1.0.0 at some point soon. cc: @cmwaters

**Review by rootulp — APPROVED** [review]
Still LGTM. Breaking changes seem isolated to the testground code.
